runtime/io.h

```diff
@@ -0,0 +1,207 @@
+
+// Internal namelist I/O
+Cookie IONAME(BeginInternalNamelistOutput)(const Descriptor &,
+    const NamelistGroup &, void **scratchArea = nullptr,
+    std::size_t scratchBytes = 0);
+Cookie IONAME(BeginInternalNamelistInput)(const Descriptor &,
+    const NamelistGroup &, void **scratchArea = nullptr,
+    std::size_t scratchBytes = 0);
+
+// External synchronous I/O initiation
+Cookie IONAME(BeginExternalListOutput)(ExternalUnit);
+Cookie IONAME(BeginExternalListInput)(ExternalUnit);
+Cookie IONAME(BeginExternalFormattedOutput)(
+    ExternalUnit, const char *format, std::size_t);
+Cookie IONAME(BeginExternalFormattedInput)(
+    ExternalUnit, const char *format, std::size_t);
+Cookie IONAME(BeginUnformattedOutput)(ExternalUnit);
+Cookie IONAME(BeginUnformattedInput)(ExternalUnit);
+Cookie IONAME(BeginNamelistOutput)(ExternalUnit, const NamelistGroup &);
+Cookie IONAME(BeginNamelistInput)(ExternalUnit, const NamelistGroup &);
+
+// Asynchronous I/O is supported (at most) for unformatted direct access
+// block transfers.
+AsynchronousId IONAME(BeginAsynchronousOutput)(
+    ExternalUnit, std::int64_t REC, const char *, std::size_t);
+AsynchronousId IONAME(BeginAsynchronousInput)(
+    ExternalUnit, std::int64_t REC, char *, std::size_t);
+Cookie IONAME(WaitForAsynchronousIo)(ExternalUnit, AsynchronousId);  // WAIT
+
+// Other I/O statements
+// TODO: OPEN & INQUIRE
+Cookie IONAME(BeginClose)(ExternalUnit);
+Cookie IONAME(BeginFlush)(ExternalUnit);
+Cookie IONAME(BeginBackspace)(ExternalUnit);
+Cookie IONAME(BeginEndfile)(ExternalUnit);
+Cookie IONAME(BeginRewind)(ExternalUnit);
+
+// Control list options
+void IONAME(SetAdvance)(Cookie, const char *, std::size_t);
+void IONAME(SetBlank)(Cookie, const char *, std::size_t);
+void IONAME(SetDecimal)(Cookie, const char *, std::size_t);
+void IONAME(SetDelim)(Cookie, const char *, std::size_t);
+void IONAME(SetPad)(Cookie, const char *, std::size_t);
+void IONAME(SetPos)(Cookie, std::int64_t);
+void IONAME(SetRec)(Cookie, std::int64_t);
+void IONAME(SetRound)(Cookie, const char *, std::size_t);
+void IONAME(SetSign)(Cookie, const char *, std::size_t);
+
+// Data item transfer for modes other than namelist.
+// Any item can be transferred by means of a descriptor; unformatted
+// transfers to/from contiguous blocks can avoid the descriptor; and there
+// are specializations for the common scalar types.
+// These functions return false when the I/O statement has encountered an error
+// or end-of-file/record condition.  Once the statement has encountered
+// an error, following items will be ignored; but compiled code should
+// check for errors and avoid the following items when a READ statement
+// might have data dependences between items, as in
+//   READ(*,ERR=666) N, (A(J),J=1,N)
+// or when input/output items might raise a spurious fault, as in
+//   WRITE(*,ERR=666) X, CRASHIFCALLEDANDERRORPRESENT()
+bool IONAME(OutputDescriptor)(Cookie, const Descriptor &);
```

Comment:
Now that the possibility is available, might it be more useful to return an ErrorHandlers (sic) here?

Reply:
No.